app/classes/query/modules/ordering.rb

```diff
@@ -1,121 +1,219 @@
 # frozen_string_literal: true
 
-module Query::Modules::Ordering
-  def initialize_order
-    by = params[:by]
-    # Let queries define custom order spec in "order", but have explicitly
-    # passed-in "by" parameter take precedence.  If neither is given, then
-    # fall back on the "default_order" finally.
-    return unless by || order.blank?
-
-    by ||= default_order
-    by = by.dup
-    reverse = !!by.sub!(/^reverse_/, "")
-    result = initialize_order_specs(by)
-    self.order = reverse ? reverse_order(result) : result
-  end
+module Query
+  module Modules
+    module Ordering
+      def initialize_order
+        by = params[:by]
+        # Let queries define custom order spec in "order", but have explicitly
+        # passed-in "by" parameter take precedence.  If neither is given, then
+        # fall back on the "default_order" finally.
+        return unless by || order.blank?
+
+        by ||= default_order
+        by = by.dup
+        reverse = !!by.sub!(/^reverse_/, "")
+        result = initialize_order_specs(by)
+        self.order = reverse ? reverse_order(result) : result
+      end
+
+      def initialize_order_specs(by)
+        sorting_method = "sort_by_#{by}"
+        unless ::Query::Modules::Ordering.private_method_defined?(
+          sorting_method
+        )
+          raise(
+            "Can't figure out how to sort #{model.name.pluralize} by :#{by}."
+          )
+        end
+
+        send(sorting_method, model)
+      end
+
+      #######
+
+      private
+
```

Comment:
Question: Should these methods be in alphabetical order?
That would make it easier to find them in the future.
(The are currently in an order that facilitates comparison with the prior version.)